### PR TITLE
Added string escape for DemuxConcatArgument

### PR DESCRIPTION
**Pull request by JKamsker**
Fixes issue when source files have a ``'`` in their path

Edit alt: Implemented https://ffmpeg.org/ffmpeg-utils.html#Quoting-and-escaping

### CODE REVIEW

File: FFMpegCore/FFMpeg/Arguments/DemuxConcatArgument.cs [hunk 1]
```diff
@@ -16,8 +16,17 @@ public class DemuxConcatArgument : IInputArgument
...
+            Values = values.Select(value => $"file '{Escape(value)}'");
         }
+
+        /// <summary>
+        /// Thanks slhck
+        /// https://superuser.com/a/787651/1089628
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        private string Escape(string value) => value.Replace("'", @"'\''");
```
Comment (rosenbjerg) [inline FFMpegCore/FFMpeg/Arguments/DemuxConcatArgument.cs:28]: Looks like you are inserting two unespaced single quotes when replacing here

**Comment by rosenbjerg** [comment]
Please add a unit test

**Comment by JKamsker** [comment]
> Please add a unit test

Not sure how i should unit test that ffmpeg actually takes this. Any advice?

File: FFMpegCore/FFMpeg/Arguments/DemuxConcatArgument.cs (hunk 1 above)
Comment (JKamsker) [reply FFMpegCore/FFMpeg/Arguments/DemuxConcatArgument.cs:28]: Works on my machine where the source has a ``'`` in its path.
Sln form https://superuser.com/a/787651/1089628

Comment (JKamsker) [reply FFMpegCore/FFMpeg/Arguments/DemuxConcatArgument.cs:28]: @rosenbjerg 
``file 'Heaven's River\05 - Investigation.m4b'``
Becomes
``file 'Heaven'\''s River\05 - Investigation.m4b'``

**Comment by JKamsker** [comment]
Added unit test

**Comment by rosenbjerg** [comment]
Nice. Thanks for the contribution :)

**Comment by rosenbjerg** [comment]
@JKamsker I have released v. 4.7.0, which contains this fix. Thanks again :)

**Pull request merged**